src/utils/stat.hpp

```diff
@@ -54,4 +58,20 @@ inline uint64_t GetMemoryUsage() {
   return memory;
 }
 
+/// Returns the size of vm.max_map_count
+inline std::optional<uint64_t> GetVmMaxMapCount() {
```

Comment:
I don't like mixing of the uin64_t and int64_t.
We are returning here the unsigned and then casting it to signed and also mixing with the default value which is a signed as well.
We should just return a signed value here in my opinion.

Reply:
Fixed it